mac

```diff
@@ -89,132 +117,192 @@ gem_install_or_update() {
   fi
 }
 
-if ! command -v brew >/dev/null; then
-  fancy_echo "Installing Homebrew ..."
-    curl -fsS \
-      'https://raw.githubusercontent.com/Homebrew/install/master/install' | ruby
+install_rvm() {
+  fancy_echo "Installing RVM"
+  curl -sSL https://rvm.io/mpapis.asc | gpg2 --import
+  curl -L https://get.rvm.io | bash -s stable
 
-    append_to_zshrc '# recommended by brew doctor'
+  source "$HOME/.rvm/scripts/rvm"
+  rvm use ruby --install --default
 
-    # shellcheck disable=SC2016
-    append_to_zshrc 'export PATH="/usr/local/bin:$PATH"' 1
+  gem update --system
+  gem_install_or_update "bundler"
+  number_of_cores=$(sysctl -n hw.ncpu)
+  bundle config --global jobs $((number_of_cores - 1))
+  fancy_echo "Installed RVM"
+}
 
-    export PATH="/usr/local/bin:$PATH"
-fi
+install_homebrew() {
+  if ! command -v brew >/dev/null; then
+    fancy_echo "Installing Homebrew ..."
+      curl -fsS \
+        'https://raw.githubusercontent.com/Homebrew/install/master/install' | ruby
 
-if brew list | grep -Fq brew-cask; then
-  fancy_echo "Uninstalling old Homebrew-Cask ..."
-  brew uninstall --force brew-cask
-fi
+      append_to_zshrc '# recommended by brew doctor'
 
-fancy_echo "Updating Homebrew formulae ..."
-brew update --force # https://github.com/Homebrew/brew/issues/1151
-brew bundle --file=- <<EOF
-tap "homebrew/services"
-tap "universal-ctags/universal-ctags"
-tap "heroku/brew"
+      # shellcheck disable=SC2016
+      append_to_zshrc 'export PATH="/usr/local/bin:$PATH"' 1
 
-# Unix
-brew "universal-ctags", args: ["HEAD"]
-brew "git"
-brew "openssl"
-brew "reattach-to-user-namespace"
-brew "the_silver_searcher"
-brew "tmux"
-brew "vim"
-brew "watchman"
-brew "gpg2"
-brew "zsh"
+      export PATH="/usr/local/bin:$PATH"
+  fi
 
-# Heroku
-brew "heroku/brew/heroku"
+  sudo chown -R $(whoami) $(brew --prefix)/*
 
-# GitHub
-brew "hub"
+  if brew list | grep -Fq brew-cask; then
+    fancy_echo "Uninstalling old Homebrew-Cask ..."
+    brew uninstall --force brew-cask
+  fi
+}
+
+append_general_dependencies() {
+  fancy_echo "Appending general dependencies to Brewfile"
+
+  tee "$HOME/.Brewfile" <<-EOF
+    # General
+    tap "thoughtbot/formulae"
+    tap "homebrew/services"
+    tap "universal-ctags/universal-ctags"
+    tap "github/gh"
+
+    # mas-cli to install macOS apps
+    brew "mas"
+
+    # General apps
+    cask "keybase"
+    cask "slack"
+    cask "google-chrome"
+    cask "1password"
+    cask "1password-cli"
+    cask "skitch"
+    cask "postman"
+    cask "iterm2"
+
+    # Unix
+    brew "universal-ctags", args: ["HEAD"]
+    brew "git"
+    brew "openssl"
+    brew "the_silver_searcher"
```

Comment:
Do we need `the_silver_searcher/watchman`? @olivierobert 
Can we remove them?
CC @suho

Reply:
It can be removed 😃